### PR TITLE
Add engine comparison table

### PR DESCRIPTION
**Pull request by FedericoCeratto**
A simple comparison table to help select the right engine based on user needs.

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 23cdc73413e7aa4dafd0109c6f2370248033926f with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62033/23cdc73413e7aa4dafd0109c6f2370248033926f/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: docs/en/engines/table-engines/index.md [hunk 1]
```diff
@@ -79,6 +79,16 @@ Engines in the family:
 - [Buffer](../../engines/table-engines/special/buffer.md#buffer)
 - [KeeperMap](../../engines/table-engines/special/keepermap.md)
 
+### Engine summary
```
Comment (rschu1ze) [inline docs/en/engines/table-engines/index.md:82]: Just `Summary`

File: docs/en/engines/table-engines/index.md [hunk 2]
```diff
@@ -79,6 +79,16 @@ Engines in the family:
...
+
+The following comparison table is not meant to be exhaustive.
+
```
Comment (rschu1ze) [inline docs/en/engines/table-engines/index.md:85]: Suggest to switch x and y axis. There are many more table engines than features.

File: docs/en/engines/table-engines/index.md [hunk 3]
```diff
@@ -79,6 +79,16 @@ Engines in the family:
...
+| Features                  | MergeTree          | ReplicatedMergeTree | Log             | Memory          | KeeperMap         | RocksDB          |
+|---------------------------|--------------------|---------------------|-----------------|-----------------|-------------------|------------------|
+| Replication               | Local              | Replicated          | Local           | Local           | Replicated        | Local            |
```
Comment (rschu1ze) [inline docs/en/engines/table-engines/index.md:88]: MergeTree, Log, Memory and RocksDB don't have "replication" (in the sense of data replicate across cluster nodes) at all.

File: docs/en/engines/table-engines/index.md [hunk 4]
```diff
@@ -79,6 +79,16 @@ Engines in the family:
...
+| Consistency               | -                  | Eventual            | Eventual        | -               | Strictly          | -                |
```
Comment (rschu1ze) [inline docs/en/engines/table-engines/index.md:89]: Strictly --> Strict

File: docs/en/engines/table-engines/index.md [hunk 5]
```diff
@@ -79,6 +79,16 @@ Engines in the family:
...
+| Use                       | OLTP               | OLTP                | Logs            | OLTP            | key-value         | key-value        |
```
Comment (rschu1ze) [inline docs/en/engines/table-engines/index.md:90]: OLTP --> OLAP

**Pull request closed without merging**